src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/ModuleBookBot.kt

```diff
@@ -0,0 +1,202 @@
+package net.ccbluex.liquidbounce.features.module.modules.misc
+
+import net.ccbluex.liquidbounce.config.types.Choice
+import net.ccbluex.liquidbounce.config.types.ChoiceConfigurable
+import net.ccbluex.liquidbounce.config.types.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.tickHandler
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.ClientModule
+import net.ccbluex.liquidbounce.features.module.modules.player.invcleaner.HotbarItemSlot
+import net.ccbluex.liquidbounce.features.module.modules.player.invcleaner.performSwap
+import net.ccbluex.liquidbounce.utils.item.findInventorySlot
+import net.minecraft.component.DataComponentTypes
+import net.minecraft.component.type.WrittenBookContentComponent
+import net.minecraft.item.Items
+import net.minecraft.network.packet.c2s.play.BookUpdateC2SPacket
+import net.minecraft.text.RawFilteredPair
+import net.minecraft.text.Style
+import net.minecraft.text.Text
+import java.util.*
+
+/**
+ * ModuleBookBot
+ *
+ * This module simplifies the process of filling and creating books using various principles,
+ * enabling efficient generation and potential automation for mass book creation or "spam."
+ *
+ * @author sqlerrorthing
+ * @since 12/28/2024
+ **/
+object ModuleBookBot : ClientModule("BookBot", Category.MISC, disableOnQuit = true) {
+    val generationMode =
+        choices(
+            "Mode",
+            RandomGenerationMode,
+            arrayOf(
+                RandomGenerationMode,
+            ),
+        ).apply { tagBy(this) }
+
+    private object Sign : ToggleableConfigurable(ModuleBookBot, "Sign", true) {
+        val bookName by text("Name", "Generated book #%count%")
+    }
+
+    init {
+        treeAll(Sign)
+    }
+
+    private val delay by int("Delay", 20, 0..200)
+
+    private var bookCount = 0
+
+    internal var random: Random = Random()
+        private set
+
+    override fun enable() {
+        bookCount = 0
+        random = Random()
+    }
+
+    @Suppress("unused")
+    private val gameTickHandler = tickHandler {
+        val book = findInventorySlot {
+            val component = it.get(DataComponentTypes.WRITABLE_BOOK_CONTENT) ?: return@findInventorySlot false
+            return@findInventorySlot it.item == Items.WRITABLE_BOOK && component.pages.isEmpty()
+        } ?: run {
+            enabled = false
+            return@tickHandler
+        }
+
+        book.performSwap(to = HotbarItemSlot(player.inventory.selectedSlot)).performAction()
```

Comment:
We should use ScheduleInventoryActionEvent for cases like this and allow to put inventory constraints.

Reply:
how?

Reply:
Listen for ScheduleInventoryActionEvent and then you can call `event.schedule(...)` - this will prevent conflicts with other modules and also automatically implements handling of `inventoryConstraints`, which can be defined as followed:
`private val inventoryConstraints = tree(PlayerInventoryConstraints())`

Reply:
and how after moving a book into hands call `writeBook` n wait for few ticks (delay)?

Reply:
resolved